### DISCUSSION TITLE
handler for token renewal

### DISCUSSION THREAD

**Discussion by dgregoire** [ahmedsakr/wstrade-api#95] [Ideas]
What would you think of having a `on` handler for when the module renews a token? Different use cases for this, the most obvious one (I think) being some sort of wired reaction - such as updating some data in some database.

**Comment by ahmedsakr** [comment]
I think it's very appropriate now that the library has implicit refreshing of tokens (if left enabled through the `config` module). Should be simple enough to support too.

**Reply by dgregoire** [reply]
@ahmedsakr  I can PR this one once you're done with #93

**Reply by ahmedsakr** [reply]
@dgregoire  Amazing, that'd be great! I wanted to ask you: how do we plan on "identifying" which tokens were renewed? The library does not store any information about the session other than the tokens, and I presume you would want to have logic in the `refresh` handler that would update the session information for the specific user.

**Reply by dgregoire** [reply]
I was holding off on #93 before I look at it more seriously, assuming you're going to make significant changes - probably something like register a `on` closure for each new client()?
